### ISSUE TITLE
False positive on `Style/ItBlockParameter`

### ISSUE THREAD

**Issue by rmm5t** [rubocop/rubocop#14567]
`Style/ItBlockParameter` prefers no `it` block parameters for multiline blocks. However, it's alerting offenses for single line blocks.

--------

## Expected behavior

A single line block with an `it` block parameter should not alert an offense.

## Actual behavior

Single line blocks that succeed a multiline construct alert a `Style/ItBlockParameter` offense.

This problem only started in rubocop v1.81.0.  It did not exist in rubocop v1.80.2.

## Steps to reproduce the problem

Here's an example code snippet that exhibits the problem. Note that the block itself is still a one-liner.

```ruby
  ADDRESS_FORMATS = {
    BTC: /\A([13][a-km-zA-HJ-NP-Z1-9]{25,34}|bc1[ac-hj-np-z02-9]{11,71})\z/,
    ETH: /\A0x[a-fA-F0-9]{40}\z/,
    XRP: /\A(r[1-9A-HJ-NP-Za-km-z]{24,34}|[XT][1-9A-HJ-NP-Za-km-z]{46,48})\z/,
  }.tap { it.default = /\A.*\z/ }.freeze
```

And the rubocop offense:

```
app/models/address.rb:15:21: C: Style/ItBlockParameter: Avoid using it block parameter for multi-line blocks.
  ADDRESS_FORMATS = { ...
                    ^
```

## RuboCop version

```
$ bundle exec rubocop -V
1.81.0 (using Parser 3.3.9.0, Prism 1.5.1, rubocop-ast 1.47.1, analyzing as Ruby 3.4, running on ruby 3.4.5) [arm64-darwin24]
  - rubocop-capybara 2.22.1
  - rubocop-factory_bot 2.27.1
  - rubocop-faker 1.3.0
  - rubocop-md 2.0.2
  - rubocop-minitest 0.38.2
  - rubocop-packaging 0.6.0
  - rubocop-performance 1.26.0
  - rubocop-rails 2.33.3
```

**Comment by dvandersluis** [comment]
A fix for this was released shortly ago in v1.81.1 (#14565).
